internal/tools/api_resources.go

```diff
@@ -0,0 +1,260 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package tools
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/asgardeo/go/pkg/api_resource"
+	"github.com/asgardeo/mcp/internal/asgardeo"
+	"github.com/mark3labs/mcp-go/mcp"
+	"github.com/mark3labs/mcp-go/server"
+)
+
+func GetListAPIResourcesTool() (mcp.Tool, server.ToolHandlerFunc) {
+	client, err := asgardeo.GetClientInstance(context.Background())
+	if err != nil {
+		log.Printf("Error initializing client instance: %v", err)
+	}
+
+	apiResourceListTool := mcp.NewTool("list_api_resources",
+		mcp.WithDescription("List API Resources registered in Asgardeo"),
+		mcp.WithString("filter",
+			mcp.Description(`Filter expression to apply, e.g., name eq Payments API`),
+		),
+		mcp.WithString("before",
+			mcp.Description(`The before cursor to use for pagination. The API will return results before this cursor`),
+		),
+		mcp.WithString("after",
+			mcp.Description(`The after cursor to use for pagination. The API will return results after this cursor`),
+		),
+		mcp.WithNumber("limit",
+			mcp.Description(`The maximum number of results to return.`),
+		),
+	)
+
+	var apiResourceListToolImpl server.ToolHandlerFunc
+	apiResourceListToolImpl = func(ctx context.Context, req mcp.CallToolRequest) (*mcp.CallToolResult, error) {
+		args := req.Params.Arguments
+		limit := getOptionalParam[int](args, "limit")
+		filter := getOptionalParam[string](args, "filter")
+		before := getOptionalParam[string](args, "before")
+		after := getOptionalParam[string](args, "after")
+		params := api_resource.GetAPIResourcesParams{
+			Limit:  limit,
+			Filter: filter,
+			Before: before,
+			After:  after,
+		}
+		resp, err := client.APIResource.List(ctx, &params)
+		if err != nil {
+			log.Printf("Error listing api resources: %v", err)
+			return nil, err
+		}
+
+		api_resources := []interface{}{}
+		for _, apiResource := range *resp.APIResources {
+			apiResourceMap := map[string]interface{}{
+				"id":   apiResource.Id,
+				"name": apiResource.Name,
+			}
+			if apiResource.Type != nil {
+				apiResourceMap["type"] = *apiResource.Type
+			}
+			if apiResource.RequiresAuthorization != nil {
+				apiResourceMap["requiresAuthorization"] = *apiResource.RequiresAuthorization
+			}
+			api_resources = append(api_resources, apiResourceMap)
+		}
+
+		return mcp.NewToolResultText(fmt.Sprintf("%+v", api_resources)), nil
+	}
+
+	return apiResourceListTool, apiResourceListToolImpl
+}
+
+func GetSearchAPIResourcesByNameTool() (mcp.Tool, server.ToolHandlerFunc) {
+	client, err := asgardeo.GetClientInstance(context.Background())
+	if err != nil {
+		log.Printf("Error initializing client instance: %v", err)
+	}
+	apiResourceGetByNameTool := mcp.NewTool("list_api_resources_by_name",
+		mcp.WithDescription("List API Resources registered in Asgardeo by name"),
```

Comment:
if this is a filtering, can we update the description similar to : `List API resources registered in Asgardeo that match a given name filter.` something like that

Reply:
Fixed with https://github.com/asgardeo/asgardeo-mcp-server/pull/7/commits/5b8a0b0a2af21b8443ae1ad324f416229caa39a8